### PR TITLE
GCP: Add install-config.yaml customization doc & document cluster isolation

### PR DESCRIPTION
**Pull request by patrickdillon**
This PR creates a standard customization.md for GCP platform. It also attempts to document cluster isolation when multiple networks are installed in the same VPC ([CORS-1217](https://jira.coreos.com/browse/CORS-1217)).

/assign @jhixson74

### CODE REVIEW

**Comment by patrickdillon** [comment]
woops already had azure on the brain as that is what I am moving to next
 /unassign @jhixson74
/assign @jstuever

File: docs/user/gcp/customization.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# GCP Platform Customization
+
+Beyond the [platform-agnostic `install-config.yaml` properties](../customization.md#platform-customization), the installer supports additional, GCP-specific properties.
+
+* `projectID` (required string): The project where the cluster should be created.
+* `region` (required string): The GCP region where the cluster should be created.
+* `network` (optional string): The ID of an existing GCP VPC where the cluster infrastructure should be provisioned.
```
Comment (abhinavdahiya) [inline docs/user/gcp/customization.md]: is it really an ID like `/project/<>/network/<>` or is it only the name of the network?

File: docs/user/gcp/customization.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+* `defaultMachinePlatform` (optional object): Default [GCP-specific machine pool properties](#machine-pools) which apply to [machine pools](../customization.md#machine-pools) that do not define their own GCP-specific properties.
+
+## Machine pools
+
+* `type` (optional string): The [GCP machine type][machine-type].
+* `zones` (optional array of strings): The availability zones used for machines in the pool.
+
+## Installing to Existing Networks & Subnetworks
+
+The installer can use an existing VPC and subnets when provisioning an OpenShift cluster. If one of `network`, `controlPlaneSubnet`, or `computeSubnet` is specified, all must be specified. The installer will use these existing networks when creating infrastructure such as VM instances, load balancers, firewall rules, and DNS zones.
```
Comment (abhinavdahiya) [inline docs/user/gcp/customization.md]: with the current validation setup, the requirement is that these should in the same project as the cluster.

Comment (abhinavdahiya) [inline docs/user/gcp/customization.md]: would like to see an install-config.yaml example included.

Comment (abhinavdahiya) [reply docs/user/gcp/customization.md]: or just link to the example below.

File: docs/user/gcp/customization.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### Cluster Isolation
+
+In a scenario where multiple clusters are installed to the same VPC network, the installer maintains cluster isolation by using firewall rules which specify allowed sources and destinations through network tags. By tagging each Compute Engine VM instance with a unique cluster id and creating corresponding firewall rules, the installer ensures that a cluster's control plane is only accessible by its own member nodes.
+
+By design, possible inter-cluster access is limited to:
+* The API, which is globally available with an external publishing strategy or available throughout the network in an internal publishing strategy
+* Debugging tools; i.e. ports on VM instances are open to the `networkCidr` for SSH & ICMP
```
Comment (abhinavdahiya) [inline docs/user/gcp/customization.md]: newtorkCIDR -> machineCIDR

File: docs/user/gcp/customization.md (hunk 1 above)
Comment (patrickdillon) [reply docs/user/gcp/customization.md]: it is a name. fixed

**Pull request merged**